Templates/Android/generated/BaseEntityCollectionReferenceRequest.java.tt

```diff
@@ -39,37 +39,43 @@
                 .post(new<#=TypeName(c)#>, body);
     }
 
+<# if (c.GetFeatures().CanExpand) { #>
     /**
      * Sets the expand clause for the request
      *
      * @param value The expand clause
      * @return The updated request
      */
     public <#=ITypeCollectionReferenceRequest(c)#> expand(final String value) {
-        addQueryOption(new QueryOption("expand", value));
+        addQueryOption(new QueryOption("$expand", value));
         return (<#=TypeCollectionReferenceRequest(c)#>)this;
     }
 
+<# } #>
+<# if (c.GetFeatures().CanSelect) { #>
```

Comment:
After looking at the pattern of emitting public Foo select(string) {...} on condition of CanSelect, I think we can do better if we push this block of code into the template class and then call a function CreateSetSelectClause(string IRequestType, string requestType) which returns string.Empty if !c.Features().CanSelect() otherwise what we have here.  it would make a number of these template classes smaller, and make it much easier to identify templetes with 'non-conformist behavior'